profiles/tests/test_profiles_graphql_gdpr_api.py

```diff
@@ -569,3 +585,66 @@ def test_api_tokens_missing(user_gql_client, service_1, query_or_delete, mocker)
         executed = user_gql_client.execute(DELETE_MY_PROFILE_MUTATION)
 
     assert_match_error_code(executed, MISSING_GDPR_API_TOKEN_ERROR)
+
+
+@pytest.mark.parametrize("dry_run", [True, False])
+def test_user_can_delete_data_from_a_service(
+    user_gql_client, service_1, service_2, mocker, dry_run
+):
+    query = DELETE_SERVICE_DATA_MUTATION_TEMPLATE.substitute(
+        service_name=service_1.name, dry_run=str(dry_run).lower()
+    )
+
+    def mock_delete_gdpr_data(self, api_token, dry_run=False):
+        return True
+
+    mocked_gdpr_delete = mocker.patch.object(
+        ServiceConnection,
+        "delete_gdpr_data",
+        autospec=True,
+        side_effect=mock_delete_gdpr_data,
+    )
+    mocker.patch.object(
+        TunnistamoTokenExchange, "fetch_api_tokens", return_value=GDPR_API_TOKENS
+    )
+    profile = ProfileFactory(user=user_gql_client.user)
+    ServiceConnectionFactory(profile=profile, service=service_1)
+    ServiceConnectionFactory(profile=profile, service=service_2)
+
+    user_gql_client.execute(query)
+
+    if dry_run:
+        assert mocked_gdpr_delete.call_count == 1
+        assert [c[2]["dry_run"] for c in mocked_gdpr_delete.mock_calls] == [True]
+        assert ServiceConnection.objects.count() == 2
+    else:
+        assert mocked_gdpr_delete.call_count == 2
+        assert [c[2]["dry_run"] for c in mocked_gdpr_delete.mock_calls] == [True, False]
+        assert ServiceConnection.objects.count() == 1
```

Comment:
This should check that the remaining service connection is for `service_2`.

Reply:
Done.